bugsnag/src/main/java/com/bugsnag/Bugsnag.java

```diff
@@ -446,6 +446,9 @@ public boolean notify(Report report, Callback reportCallback) {
             return false;
         }
 
+        // Add thread metadata to the report
+        report.mergeMetaData(THREAD_METADATA.get());
```

Comment:
This will have implications on which item of metadata is used in the case of duplicate keys when merging metadata. Before the precedence was:
1. Report metadata 
1. Thread metadata
1. Global metadata

Now it will be:
1. Report metadata
1. Global metadata
1. Thread metadata

Reply:
That's correct. I think we should add the thread metadata first, so it can be used in any global level callbacks